### PR TITLE
[WIP] Reformatted and cleaned some code

### PR DESCRIPTION
**Pull request by sampreet-arthi**
- Reformatted PPO1, A2C, VPG (first 3 commits)
- Reformatted the rewards tracking (separate commit)
- Made getting observation shape and action shape consistent and the same across all files (refer to last commit)
- Moved all saving related parameters to the trainer

- [x] Saving/loading models: clean existing code, maybe #161 , #181 
- [x] Fix the evaluate function in Trainer (#164)

If there are any other changes required let me know.

### CODE REVIEW

**Comment by sampreet-arthi** [comment]
@Sharad24 @Het-Shah @ajaysub110 Let me know if this looks good. It's mostly just reformatting but a lot of it was necessary to clean the code (got confusing with reward logging and atari support needed PPO to be reformatted to use actor-critic). Made mainly so that #175 didn't have too much new unrelated code.

**Comment by sampreet-arthi** [comment]
I'm thinking of moving all saving/logging related arguments to the trainer. Currently, the Agent has `save_model` and `run_num`. It would just be cleaner (and more sound) to keep those arguments in the Trainer class instead.

**Comment by sampreet-arthi** [comment]
Also, we could remove `self.load_model` from the agent and make it part of the trainer too. The Trainer could then call `Agent.load_params()` to load some path. The logic is that keep any arguments not needed for training outside of the agent. It makes more sense to include it as part of the Trainer instead.

**Comment by Sharad24** [comment]
Please add a [WIP] tag or mark the PR as draft? If its not complete

**Comment by Sharad24** [comment]
Do you think it would be a good idea to keep the issues you're facing for another PR (or separate PRs perhaps)?

**Comment by sampreet-arthi** [comment]
Most of these changes are just reformatting or minor restructuring. I figured there would be too many PRs if they were to be solved separately. I've kept each commit sort of standalone. If you want I can open separate PRs next time, I didn't plan to do these many changes initially.

**Comment by sampreet-arthi** [comment]
@ajaysub110 Was there a reason we added `run_num` as an argument? What it's doing is just letting the user choose which run number they want to save the model as (not load). Is that required?

**Comment by sampreet-arthi** [comment]
If the Travis builds, we can merge this @Sharad24 . I'm not adding anything new in this PR.

**Comment by Sharad24** [comment]
Everything done?

**Comment by sampreet-arthi** [comment]
There's an isort-black compatibility issue now :( I'll sort that out

**Comment by Sharad24** [comment]
I'd say keep it to 4.2.5 and open a separate issue for that? Because I think there are other compatibility issues as well

File: .isort.cfg [hunk 1]
```diff
@@ -1,2 +1,7 @@
 [settings]
 known_third_party = cv2,gym,matplotlib,numpy,pytest,scipy,setuptools,torch
+multi_line_output=3
+include_trailing_comma=True
+force_grid_wrap=0
+use_parentheses=True
+line_length=88
```
Comment (Sharad24) [inline .isort.cfg:7]: Is line length 88 still followed? Hasn't that been increased now?

(Not sure)

**Comment by sampreet-arthi** [comment]
I modified `.isort.cfg` as recommended on the official `black` website. It seems to build fine now. (lint check has passed)

File: .isort.cfg (hunk 1 above)
Comment (sampreet-arthi) [reply .isort.cfg:7]: This is what black recommended. PEP8 recommends 79, should I change that?

**Comment by sampreet-arthi** [comment]
Ready to merge @Sharad24

File: .isort.cfg (hunk 1 above)
Comment (sampreet-arthi) [reply .isort.cfg:7]: However, they do say that it's fine as long as the team is fine with it :p

**Pull request merged**